libqtile/config.py

```diff
@@ -490,28 +490,34 @@ def gaps(self) -> Iterable[BarType]:
 
     @property
     def dx(self) -> int:
-        return self.x + self.left.size if self.left else self.x
+        return (
+            self.x + self.left.size
+            if (self.left and not getattr(self.left, "floating"))
+            else self.x
+        )
```

Comment:
Why using getattr here?

Also it's most likely easier to read without a ternary:

```suggestion
        if self.left and not self.left.floating:
            return self.x + self.left.size
        return self.x
```

Reply:
Used getattr because the attribute doesn't exist for `Gap` objects.

Agreed on tidier code.